packages/commerce-sdk-react/src/hooks/types.ts

```diff
@@ -45,31 +45,15 @@ export interface QueryParams {
     shortCode?: string
 }
 
-export type QueryResponse<T> =
-    | {
-          isLoading: boolean
-          error?: undefined
-          data?: undefined
-      }
-    | {
-          isLoading: true
-          error?: Error
-          data?: T
-      }
-    | {
-          isLoading: false
-          error: Error
-          data?: undefined
-      }
-    | {
-          isLoading: false
-          error?: undefined
-          data: T
-      }
+export type QueryResponse<T> = {
+    isLoading: boolean
+    error?: Error | undefined
+    data?: T | undefined
+}
 
 // TODO: Include action name as a key in addition to "execute"?
-export type ActionResponse<A, R> = QueryResponse<R> & {
-    execute: (arg: A) => void
+export type ActionResponse<A extends any[], R> = QueryResponse<R> & {
+    execute: (...arg: A) => void
```

Comment:
How generic is `ActionResponse` intended to be? If it's just for the raw SCAPI endpoints, then we should leave this as a single argument. If there are other things (SLAS helpers?) then this is fine.

Reply:
I'm trying to make it work for raw SCAPI endpoint as well as those SLAS helpers.